### PR TITLE
dts: qcom: msm8953-xiaomi-tissot: Add some new features

### PR DESCRIPTION
**Pull request by JIaxyga**
- Add battery info
- Add sound configuration
- Add i2c-gpio sensors
- Add SimpleFB
- Sort alphabetically

### CODE REVIEW

**Comment by fekz115** [comment]
You added new drivers. Are they already enabled in your config?

File: arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts
Comment (M0Rf30) [inline arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts:111]: tissot seems to report the same sensor of vince, also on hwdeviceinfo. But I'm successfully using CONFIG_LTRF216A and this snippet in dts
```
        light-sensor@53 {
            compatible = "liteon,ltrf216a";
            reg = <0x53>;
            vdd-supply = <&pm8953_l10>;
        };
```
You should be able to obtain values with 
`cat /sys/bus/iio/devices/iio:device0/in_illuminance_raw`

File: arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts
Comment (M0Rf30) [inline arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts:76]: this requires a
`
/delete-node/ &cont_splash_mem;
`
on top of this file

**Comment by JIaxyga** [comment]
> You added new drivers. Are they already enabled in your config?

It doesn't matter. It still doesn't work correctly.

File: arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts
Comment (JIaxyga) [reply arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts:76]: true. I didn't notice that this region was added to msm8953

File: arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts
Comment (M0Rf30) [inline arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts:15]: I don't know if will work in this way. You are calling the node and override its value, If I remember correctly you can only add attributes, not override them.

File: arch/arm64/configs/xiaomi-tissot.config
Comment (fekz115) [inline arch/arm64/configs/xiaomi-tissot.config:3]: Recheck how battery working, this should be built-in

Comment (M0Rf30) [inline arch/arm64/configs/xiaomi-tissot.config:3]: to make them work I  had to set smchg and fg to =y

Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:3]: Not necessary. Charger and fg just need to be in the same state

File: arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts
Comment (JIaxyga) [reply arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts:15]: There is no point in redefining it. The address remains the same. Maintainers will definitely tell you about this when trying to upstream

Comment (M0Rf30) [reply arch/arm64/boot/dts/qcom/msm8953-xiaomi-tissot.dts:15]: Ok

File: arch/arm64/configs/xiaomi-tissot.config
Comment (M0Rf30) [reply arch/arm64/configs/xiaomi-tissot.config:3]: In case of ltrf216a don't forget to add `CONFIG_LTRF216A`

Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:3]: See comments from @fekz115. Let @alikates judge us

Comment (alikates) [reply arch/arm64/configs/xiaomi-tissot.config:3]: Yes, they can be builtin or module. Both have to be the same, but now OTG and role detection works, the usb driver depends on it too and iirc that one is builtin.

Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:3]: > Yes, they can be builtin or module. Both have to be the same, but now that OTG and role detection works, the usb driver depends on it too and iirc that one is builtin.

So what do I change to build-in?

Comment (M0Rf30) [reply arch/arm64/configs/xiaomi-tissot.config:3]: CONFIG_BATTERY_QCOM_FG=y
CONFIG_CHARGER_QCOM_SMBCHG=y

File: arch/arm64/configs/xiaomi-tissot.config
Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:6]: They will be installed by modules automatically because they depend on drm and msm_drm which are set as modules in defconfig

Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:6]: msm_drm was switched to module due to tissot as it caused kernel panic. This problem has been fixed with the latest kernel releases

Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:6]: I really don't care. No problem if @alikates wants it too

Comment (M0Rf30) [reply arch/arm64/configs/xiaomi-tissot.config:6]: > If someone decides to switch drm to build-in, it will be less painful, since it will not be necessary to enable build-in for panels again

got it, what about the light sensor?

**Review by M0Rf30 — APPROVED** [review]

File: arch/arm64/configs/xiaomi-tissot.config
Comment (JIaxyga) [reply arch/arm64/configs/xiaomi-tissot.config:6]: > > If someone decides to switch drm to build-in, it will be less painful, since it will not be necessary to enable build-in for panels again
> 
> got it, what about the light sensor?

I think we need to make some adjustments to the driver, because it still does not work correctly. We will also need to add a new compatible to the driver, because ltr579 still has some differences. I suggest not touching ltr579 in dts for now. We'll fix this in the next release

**Comment by M0Rf30** [comment]
Please edit the gpio-keys block according to this
https://github.com/msm8953-mainline/linux/pull/91#discussion_r1177628976

**Comment by JIaxyga** [comment]
> Please edit the gpio-keys block according to this [#91 (comment)](https://github.com/msm8953-mainline/linux/pull/91#discussion_r1177628976)

Tissot along with the hall sensor was upstream in this look. Upstream maintainers will make the necessary changes themselves

**Comment by JIaxyga** [comment]
> > Please edit the gpio-keys block according to this [#91 (comment)](https://github.com/msm8953-mainline/linux/pull/91#discussion_r1177628976)
> 
> Tissot along with the hall sensor was upstream in this look. Upstream maintainers will make the necessary changes themselves

There is no point in making changes here as it will make conflicts in future

**Comment by JIaxyga** [comment]
> > > Please edit the gpio-keys block according to this [#91 (comment)](https://github.com/msm8953-mainline/linux/pull/91#discussion_r1177628976)
> > 
> > 
> > Tissot along with the hall sensor was upstream in this look. Upstream maintainers will make the necessary changes themselves
> 
> There is no point in making changes here as it will make conflicts in future

I suggest waiting for the moment when the upstream linking is successful

**Comment by alikates** [comment]
One small issue to fix before merging. Can you change the commit message for the config changes to the format that's been used previously? (arm64: config: ...)

**Comment by JIaxyga** [comment]
> One small issue to fix before merging. Can you change the commit message for the config changes to the format that's been used previously? (arm64: config: ...)

Oooooops xD

**Pull request merged**
